pkg/agent/proxy/proxier.go

```diff
@@ -1010,6 +1015,54 @@ func (p *proxier) GetServiceFlowKeys(serviceName, namespace string) ([]string, [
 	return flows, groups, found
 }
 
+func (p *proxier) HandlePacketIn(pktIn *ofctrl.PacketIn) error {
+	if pktIn == nil {
+		return errors.New("empty packetin for Antrea Proxy")
+	}
+	// Get ethernet data.
+	ethernetPkt, err := openflow.GetEthernetPacket(pktIn)
+	if err != nil {
+		return err
+	}
+	srcMAC := ethernetPkt.HWDst.String()
+	dstMAC := ethernetPkt.HWSrc.String()
+
+	var (
+		srcIP, dstIP string
+		proto        uint8
+		isIPv6       bool
+	)
+	switch ipPkt := ethernetPkt.Data.(type) {
+	case *protocol.IPv4:
+		srcIP = ipPkt.NWDst.String()
+		dstIP = ipPkt.NWSrc.String()
+		proto = ipPkt.Protocol
+		isIPv6 = false
+	case *protocol.IPv6:
+		srcIP = ipPkt.NWDst.String()
+		dstIP = ipPkt.NWSrc.String()
+		proto = ipPkt.NextHeader
+		isIPv6 = true
+	}
+
+	matches := pktIn.GetMatches()
+	if matches.GetMatchByName(binding.OxmFieldInPort) == nil {
```

Comment:
Use a variable for inPort field to avoid a second parse in line 1052?

Reply:
Done.